test/integration/AuthenticatedFileResourceStore.test.ts

```diff
@@ -0,0 +1,238 @@
+          requestUrl,
+          'POST',
+          {
+            'content-type': 'application/octet-stream',
+            slug: 'permanent.txt',
+            'transfer-encoding': 'chunked',
+          },
+          fileData,
+        );
+        expect(response.statusCode).toBe(200);
+      }
+    },
+  );
+  afterAll(
+    async(): Promise<void> => {
+      await setAcl(
+        store,
+        'http://test.com/',
+        { read: true, write: true, append: true },
+        true,
+        true,
+        true,
+        undefined,
+        'agent',
+      );
+
+      // Delete permanente file
+      const requestUrl = new URL('http://test.com/permanent.txt');
+      const response = await call(
+        handler,
+        requestUrl,
+        'DELETE',
+        {},
+        [],
+      );
+      expect(response.statusCode).toBe(200);
+
+      // Delete .acl file
+      await store.deleteResource({ path: 'http://test.com/.acl' });
+    },
+  );
+  describe('with acl', (): void => {
+    it('can add a file to the store, read it and delete it if allowed.', async(): Promise<
+    void
+    > => {
+      await setAcl(
+        store,
+        'http://test.com/',
+        { read: true, write: true, append: true },
+        true,
+        true,
+        true,
+        undefined,
+        'agent',
+      );
+
+      // POST
+      let requestUrl = new URL('http://test.com/');
+
+      const fileData = await fs.readFile('test/testfiles/testfile1.txt');
+
+      let response: MockResponse<any> = await callFile(
+        handler,
+        requestUrl,
+        'POST',
+        {
+          'content-type': 'application/octet-stream',
+          slug: 'testfile1.txt',
+          'transfer-encoding': 'chunked',
+        },
+        fileData,
+      );
+      expect(response.statusCode).toBe(200);
+      expect(response._getData()).toHaveLength(0);
+      const id = response._getHeaders().location;
+      expect(id).toContain(url.format(requestUrl));
+
+      // GET
+      requestUrl = new URL(id);
+      response = await call(
+        handler,
+        requestUrl,
+        'GET',
+        { accept: 'text/*' },
+        [],
+      );
+      expect(response.statusCode).toBe(200);
+      expect(response._getHeaders().location).toBe(id);
+      expect(response._getBuffer().toString()).toContain('TESTFILE1');
+
+      // DELETE
+      response = await call(handler, requestUrl, 'DELETE', {}, []);
+      expect(response.statusCode).toBe(200);
+      expect(response._getData()).toHaveLength(0);
+      expect(response._getHeaders().location).toBe(url.format(requestUrl));
+
+      // GET
+      response = await call(
+        handler,
+        requestUrl,
+        'GET',
+        { accept: 'text/*' },
+        [],
+      );
+      expect(response.statusCode).toBe(404);
+      expect(response._getData()).toContain('NotFoundHttpError');
+    });
+
+    it('can not add a file to the store if not allowed.', async(): Promise<
+    void
+    > => {
+      await setAcl(
```

Comment:
Too many params to this method here, no idea what all the `true`s mean.
Can we make them named params in a single options hash?